src/devices/GoPiGo3/Movements/Motor.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.GoPiGo3.Models;
+using System;
+using System.Threading;
+
+namespace Iot.Device.GoPiGo3.Movements
+{
+    /// <summary>
+    /// Polarity of the motor
+    /// </summary>
+    public enum Polarity
+    {
+#pragma warning disable
+        Backward = -1, Forward = 1, OppositeDirection = 0
+#pragma warning restore
+    };
+
+    public class Motor
+    {
+        // represent the Brick
+        private GoPiGo _goPiGo = null;
+        private int _tacho;
+        private Timer _timer = null;
+        private int _periodRefresh;
+
+        /// <summary>
+        /// Create a motor
+        /// </summary>
+        /// <param name="port">Motor port</param>
+        public Motor(GoPiGo brick, MotorPort port) : this(brick, port, 1000) { }
+
+        public Motor(GoPiGo brick, MotorPort port, int timeout)
+        {
+            if (port == MotorPort.Both)
+                throw new ArgumentException($"Motor class can only have 1 motor");
+            _goPiGo = brick;
+            Port = port;
+            _periodRefresh = timeout;
+            _timer = new Timer(UpdateSensor, this, TimeSpan.FromMilliseconds(timeout), TimeSpan.FromMilliseconds(timeout));
+        }
+
+        /// <summary>
+        /// Set the speed of the motor
+        /// </summary>
+        /// <param name="speed">speed is between -255 and +255</param>
+        public void SetSpeed(int speed)
+        {
+            speed = Math.Clamp(speed, -255, 255);
+            _goPiGo.SetMotorPower(Port, speed);
+        }
+
+        /// <summary>
+        /// Set Tachometer encoder offset
+        /// Use this to reset or setup a specific position
+        /// </summary>
+        /// <param name="position">New offset, 0 to reset</param>
+        public void SetTachoCount(int position)
```

Comment:
Is negative position valid?

Reply:
Yes, a negative position can be valid.  That means you went backward.